selectors/onboarding.ts

```diff
@@ -127,30 +109,64 @@ export const legacyNextStepsNotFinished = createSelector(
     },
 );
 
+// Loop through all Steps. For each step, check that
+export const hasLegacyNextStepsPreferences = createSelector(
+    'hasLegacyNextStepsPreferences',
+    (state: GlobalState) => getCategory(state, Preferences.RECOMMENDED_NEXT_STEPS),
+    (state: GlobalState) => getSteps(state),
+    (stepPreferences, mySteps) => {
+        const checkPref = (step: StepType) => stepPreferences.some((pref) => (pref.name === step.id));
+        return mySteps.some(checkPref);
+    },
+);
+
 export const getShowTaskListBool = createSelector(
     'getShowTaskListBool',
     (state: GlobalState) => state,
     (state: GlobalState) => getCategory(state, OnboardingTaskCategory),
     (state: GlobalState) => getCategory(state, Preferences.RECOMMENDED_NEXT_STEPS),
     (state, onboardingPreferences, legacyStepsPreferences) => {
+        const isMobileView = isMobile();
+
         // conditions to validate scenario where users (initially first_admins) had already set any of the onboarding task list preferences values.
         // We check wether the preference value exists meaning the onboarding tasks list already started no matter what the state of the process is
         const hasUserStartedOnboardingTaskListProcess = onboardingPreferences?.some((pref) =>
             pref.name === OnboardingTaskList.ONBOARDING_TASK_LIST_SHOW || pref.name === OnboardingTaskList.ONBOARDING_TASK_LIST_OPEN);
 
+        const taskListStatus = getBool(state, OnboardingTaskCategory, OnboardingTaskList.ONBOARDING_TASK_LIST_SHOW);
```

Comment:
this refactor tries to avoid to execute extra logic once the user has already got the preferences set for the new onboarding. The general logic hasn't change (the unit tests remain exactly the same), but this refactor is an improvement in the way the logic was executed.